src/compile/legend.ts

```diff
@@ -162,15 +163,26 @@ export namespace properties {
       // For non-color legend, apply color config if there is no fill / stroke config.
       // (For color, do not override scale specified!)
       symbols[filled ? 'fill' : 'stroke'] = symbols[filled ? 'fill' : 'stroke'] ||
-        {value: model.config().mark.color};
+        {value: cfg.mark.color};
     }
 
     if (legend.symbolColor !== undefined) {
       symbols.fill = {value: legend.symbolColor};
+    } else if (symbols.fill === undefined) {
+      // fall back to mark config colors for legend fill
+      if (cfg.mark.color !== undefined) {
```

Comment:
FYI, Looks like this line introduced a bug #1500 -- but I've just fixed in https://github.com/vega/vega-lite/pull/1504